subsys/bluetooth/mesh/shell.c

```diff
@@ -1721,15 +2018,94 @@ static int cmd_mod_pub(const struct shell *shell, size_t argc, char *argv[])
 	}
 }
 
+static int mod_pub_va_set(const struct shell *shell, uint16_t addr, uint16_t mod_id,
+			  uint16_t cid, char *argv[])
+{
+	struct bt_mesh_cfg_mod_pub_va pub_va;
+	uint8_t status, count;
+	uint8_t label[16];
+	uint16_t interval;
+	size_t len;
+	int err;
+
+	len = hex2bin(argv[0], strlen(argv[0]), label, sizeof(label));
+	(void)memset(label + len, 0, sizeof(label) - len);
+	pub_va.label = (const uint8_t *)&label;
+	pub_va.pub.app_idx = strtoul(argv[1], NULL, 0);
+	pub_va.pub.cred_flag = str2bool(argv[2]);
+	pub_va.pub.ttl = strtoul(argv[3], NULL, 0);
+	pub_va.pub.period = strtoul(argv[4], NULL, 0);
+
+	count = strtoul(argv[5], NULL, 0);
+	if (count > 7) {
+		shell_print(shell, "Invalid retransmit count");
+		return -EINVAL;
+	}
+	interval = strtoul(argv[6], NULL, 0);
+	if (interval > (31 * 50) || (interval % 50)) {
```

Comment:
Use rounding for the interval instead? This looks like it's going to be hard to get right without looking at the spec or source code